apps/sertra/components/PackageDataFetcher.tsx

```diff
@@ -0,0 +1,65 @@
+import { useEffect, useState } from "react";
+import { ListingMod, TSListingMod } from "../api/models";
+import { ApiURLs, TsApiURLs } from "../api/urls";
+
+// TODO: To be replaced with ServerListingDetailData when the listing will contain an actual
+// object instead of an list of strings
+
+interface TempModRefType {
+  mods: string[];
+}
+
+export const FetchListingData = async (listingId: string) => {
+  const [listing, setListing] = useState<TempModRefType>();
+  useEffect(() => {
+    async function fetchListing() {
+      return await fetch(ApiURLs.ServerDetail(listingId));
+    }
+    async function jsonifyListing() {
+      setListing(await (await fetchListing()).json());
+    }
+    jsonifyListing();
+  });
```

Comment:
Why so verbose? There's no need to define named functions like this

```ts
useEffect(() => {
  fetch(ApiURLs.ServerDetail(listingId)).then((resp) => {
    setListing(resp.json());
  });
}, [listingId]);
```

Same applies for the other `useEffect` below.